### PR TITLE
sriov: added networknodestate unit tests

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

File: pkg/sriov/networknodestate_test.go [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: nsName,
+		},
+		Spec:   srIovV1.SriovNetworkNodeStateSpec{},
+		Status: srIovV1.SriovNetworkNodeStateStatus{},
+	}
+}
+
+func buildNodeNetworkStateSyncStatus(name, nsName string, syncStatus string) *srIovV1.SriovNetworkNodeState {
```
Comment (achuzhoy) [inline pkg/sriov/networknodestate_test.go:367]: nit:
in function buildNodeNetworkState above you specify separately the type for name and nsName.
In this function you "save" one time writing the type, but still separate for syncStatus. No consistency.

**Review by sebrandon1 — COMMENTED** [review]
Do these `err` checks need test cases?

File: pkg/sriov/networknodestate_test.go [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+
+		networkNodeState := buildNodeNetworkStateWithNics(testCase.netInterface)
+		runtimeObjects = append(runtimeObjects, networkNodeState)
+
+		testSettings = clients.GetTestClients(runtimeObjects)
+
+		networkNodeStateBuilder := NewNetworkNodeStateBuilder(testSettings, defaultNodeName, defaultNodeNsName)
+		nics, err := networkNodeStateBuilder.GetNICs()
+
+		assert.Nil(t, err)
```
[the remark this answers is not in the document]
Comment (kononovn) [reply pkg/sriov/networknodestate_test.go:160]: Added additional test case

File: pkg/sriov/networknodestate_test.go [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+		},
+	}
+
+	for _, testCase := range testCases {
+		networkNodeState := buildNodeNetworkStateWithNics(testCase.netInterface)
+		testSettings := clients.GetTestClients(append([]runtime.Object{}, networkNodeState))
+		networkNodeStateBuilder := NewNetworkNodeStateBuilder(testSettings, defaultNodeName, defaultNodeNsName)
+		nics, err := networkNodeStateBuilder.GetUpNICs()
+
+		assert.Nil(t, err)
```
[the remark this answers is not in the document]
Comment (kononovn) [reply pkg/sriov/networknodestate_test.go:220]: Added additional test case

**Review by sebrandon1 — APPROVED** [review]

File: pkg/sriov/networknodestate_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sebrandon1) [reply pkg/sriov/networknodestate_test.go]: Thanks!

**Review by achuzhoy — APPROVED** [review]

**Pull request merged**